datacube/utils/xarray_geoextensions.py

```diff
@@ -32,8 +32,21 @@ def _xarray_affine(obj):
 
     dims = crs.dimensions
 
-    xres, xoff = data_resolution_and_offset(obj[dims[1]].values)
-    yres, yoff = data_resolution_and_offset(obj[dims[0]].values)
+    xres = data_resolution(obj[dims[1]].values)
+    yres = data_resolution(obj[dims[0]].values)
+    if xres is None and yres is None:
+        try:
+            xres, yres = obj.resolution
+        except:
+            raise ValueError('Resolution is missing')
+    elif xres is None:
+        # assume the pixel is square
+        xres = -yres
+    else:
+        yres = -xres
```

Comment:
this logic is wrong, you are are overwriting valid yres with -xres. I would deal with each axis separately

1. Use data derived resolution
2. Fallback to `resolution` attribute
3. Fallback to resolution along other axis if available
4. Raise error

Reply:
Totally agree with the wrong logic comment. Got a question about the priority though. Do we want to assume square pixel and use the other axis first or do we want to get the info from attribute? I got an expression that only when a single pixel we want to read the attribute, otherwise using the other axis.

Reply:
@Kirill888 does this mean you are OK with this? :smile:

Reply:
How about just 1, 2, 4? If we can get `resolution` attached always.

Reply:
@emmaai I'd say that if resolution is recorded it should be used for `<2` element axis, only if it's missing we can fallback to assuming square pixels with the negative sign for y-axis

@uchchwhash unlike previous change this only fallbacks to resolution attribute, not depends on it, so it will do wrong thing less often

Reply:
@emmaai didn't read your comment right last time. Skipping 3 is fine with me, primarily because assuming square pixels is kinda ok, but we also need to know which way is up and that's a less obvious choice to make. So not trying to guess resolution if not recorded is probably best thing to do.